src/components/ha-switch.ts

```diff
@@ -55,6 +80,12 @@ export class HaSwitch extends MwcSwitch {
       `,
     ];
   }
+
+  private _haChangeHandler(e: Event) {
+    this.mdcFoundation.handleChange(e);
+    // catch "click" event and sync properties
+    this.checked = this.formElement.checked;
```

Comment:
We shouldn't do this. This will lead to nasty bugs.

When a parent renders this element, it will diff against itself to know if it needs to change the property. 

So let's say this happens:
 - Parent renders checked = true
 - User changes it, sets checked=false
 - Parent renders checked = true -> this will not update `<ha-switch>` because it already rendered `true` last time, even though the property current value is `false`

Reply:
This was actually a copy/paste from `mwc-switch`, because it was a private function

Reply:
It would be weird of the parent not to update it's value btw when the switch is changed.

Reply:
Also, there is no `value-changed` event, just the normal change event of the native checkbox, so it would be impossible for the parent to see what the state is when we would not update the prop? `ev.target.checked` would no longer work.

I could rewrite it and fire the `value-changed` event and not do this, but that would differ us a lot from `mwc-switch` and means a lot of change.

Reply:
ok let's keep it.